File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution(object):
    def longestOnes(self, nums, k):
        left = 0
        right = 0
        result = 0
        zero_counter = 0
        while (right < len(nums)):
            if nums[right] == 0:
                zero_counter += 1
            
            while zero_counter > k:
                if nums[left] == 0:
                    zero_counter -= 1
                left += 1
            result = max(result, right-left+1)
            right += 1
         
        return result      
        

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.js
var longestOnes = function(nums, k) {
    let left = 0
    let right = 0
    let result = 0
    let zero_counter = 0
    while (right<nums.length){
        if (nums[right] == 0){
            zero_counter += 1
        }       
        while(zero_counter > k){
            if (nums[left] == 0){
                zero_counter -= 1
            }     
            left += 1
        }
        result = Math.max(result, right-left+1)
        right += 1
    }    
    return result      
        
};